controller_interface/test/test_controller_with_options.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 ros2_control development team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <string>
+#include "test_controller_with_options.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace controller_with_options
+{
+controller_interface::return_type ControllerWithOptions::init(
```

Comment:
Why not move controllers implementations into header file so that cpp has only tests inside. @bmagyar what do you think?

Reply:
done